src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create new book in DB, book = " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = getBookFromResultSet(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = getBookFromResultSet(resultSet);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find any books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new SQLException("Expected to effect at least one row, but effected 0");
+            } else {
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book = " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new SQLException("Expected to effect at least one row, but effected 0");
+            }
+            return true;

```

Comment:
Instead of returning `true` all the time in the `deleteById` method, let's return a boolean value depending on `preparedStatement.executeUpdate()` result. Use `return affectedRows > 0;` instead of `return true;`.